File: example-microservice-java/src/main/java/com/example/java/customer/repository/CustomerRepository.java
package com.example.java.customer.repository;

import com.example.java.customer.service.CustomerModel;
import com.example.java.customer.api.dto.CustomerCreateDto;
import com.example.java.customer.api.dto.CustomerUpdateDto;

import java.util.Optional;
import java.util.UUID;

public interface CustomerRepository {

    Optional<CustomerModel> getCustomer(UUID id);

    CustomerModel createCustomer(CustomerCreateDto newCustomer);

    CustomerModel updateCustomer(UUID id, CustomerUpdateDto updatedCustomer);

    void deleteCustomer(UUID id);

}

File: example-microservice-kotlin/src/main/kotlin/com/gooddata/kotlin/customer/repository/CustomerRepository.kt
package com.example.kotlin.customer.repository

import com.example.kotlin.customer.api.dto.CustomerCreateDto
import com.example.kotlin.customer.api.dto.CustomerUpdateDto
import com.example.kotlin.customer.service.CustomerModel
import java.util.*

interface CustomerRepository {

    fun getCustomer(id: UUID): CustomerModel?

    fun createCustomer(newCustomer: CustomerCreateDto): CustomerModel

    fun updateCustomer(id: UUID, updatedCustomer: CustomerUpdateDto): CustomerModel

    fun deleteCustomer(id: UUID)

}
